### PR TITLE
[Network]Support p2s vpn gateway and vpn server configuration

### PR DESCRIPTION
**Pull request by MyronFanQiu**
---

Fix Azure/azure-cli#13020

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S171

File: src/virtual-wan/azext_vwan/_help.py [hunk 1]
```diff
@@ -297,3 +297,102 @@
...
+
+# region VpnServerConfig
+helps['network vpn-server-config'] = """
+    type: group
+    short-summary: Manage VPN server configuration.
+"""
+
+helps['network vpn-server-config create'] = """
+    type: command
+    short-summary: Create a VPN server configuration.
```
Comment (jsntcy) [inline src/virtual-wan/azext_vwan/_help.py:309]: Add examples for command help

File: src/virtual-wan/azext_vwan/_params.py [hunk 2]
```diff
@@ -149,3 +153,44 @@ def load_arguments(self, _):
...
+        c.argument('sa_data_size_kilobytes', options_list='--sa-data-size', help='IPSec Security Association (also called Quick Mode or Phase 2 SA) payload size in KB for a site-to-site VPN tunnel.', type=int)
+        c.argument('ipsec_encryption', arg_type=get_enum_type(IpsecEncryption), help='IPSec encryption algorithm (IKE phase 1).')
+        c.argument('ipsec_integrity', arg_type=get_enum_type(IpsecIntegrity), help='IPSec integrity algorithm (IKE phase 1).')
+        c.argument('ike_encryption', arg_type=get_enum_type(IkeEncryption), help='IKE encryption algorithm (IKE phase 2).')
+        c.argument('ike_integrity', arg_type=get_enum_type(IkeIntegrity), help='IKE integrity algorithm (IKE phase 2).')
+        c.argument('dh_group', arg_type=get_enum_type(DhGroup), help='DH Groups used in IKE Phase 1 for initial SA.')
+        c.argument('pfs_group', arg_type=get_enum_type(PfsGroup), help='The Pfs Groups used in IKE Phase 2 for new child SA.')
+        c.argument('index', type=int, help='List index of the item (starting with 0).')
```
Comment (jsntcy) [inline src/virtual-wan/azext_vwan/_params.py:184]: What does the item mean?

File: pylintrc [hunk 3]
```diff
@@ -28,7 +28,7 @@ disable=missing-docstring,
 
 [TYPECHECK]
 # For Azure CLI extensions, we ignore some import errors as they'll be available in the environment of the CLI
-ignored-modules=azure,azure.cli,azure.cli.core,azure.cli.core.commands,knack,msrestazure,argcomplete,azure_devtools,isodate
+ignored-modules=azure,azure.cli,azure.cli.core,azure.cli.core.commands,knack,msrestazure,argcomplete,azure_devtools,isodate,OpenSSL
```
Comment (haroldrandom) [inline pylintrc:31]: Why need OpenSSL to crypto?

File: src/virtual-wan/azext_vwan/tests/latest/ApplicationGatewayAuthCert.pem [hunk 4]
```diff
@@ -0,0 +1,20 @@
+-----BEGIN CERTIFICATE-----
```
Comment (haroldrandom) [inline src/virtual-wan/azext_vwan/tests/latest/ApplicationGatewayAuthCert.pem:1]: better to move it to another dedicated folder named with `data` or `fixture`

File: src/virtual-wan/azext_vwan/tests/latest/ApplicationGatewayAuthCert.pem [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+7+5QpLxSKc2478J+PvTwy/rfWKJLbEImoM7WE3imWgrHUNMCAwEAAaNlMGMwYQYD
+VR0BBFowWIAQxDXqD2PYxSlriJ6HuzZ+b6EyMDAxLjAsBgNVBAMTJUFwcGxpY2F0
+aW9uR2F0ZXdheVNzbENlcnQtVW5pdFRlc3QtMDGCEI+aE5UeqMidTZV/Vm/Whcow
+DQYJKoZIhvcNAQELBQADggEBADSC6PHkVYmoVOyyW2g7q3GG3MmXa43/HQNTM+pT
+3HENAlhGg8T9EFrOQ/0QAoEQbrtsP8Isj5u2Bd4MOlYA180ExyfL/NFb7QpXq/pR
+LbePMQAQicdSvk/9RyrNRBfNjT2KNlBp3RZtCkJjQWlc5ylWydPy4JPUZxit/5yg
+CRT+cmN1ioF469Ki8hzBlKmrDg8jkFUwFyD7B0fEfPP0gSZI3F6HqgCK2hzBTjuj
+ZjddZyn2XxXlxLu57C83CNPP8fbsGW2CwolgmYvt+dJsdDnT7VKX5RS4cOPp/JGG
+7QdhzqVBoK0vZP6gHEAOPoqBVUWf8SmlVuAVGmDl8EjLnK8=
+-----END CERTIFICATE-----
```
Comment (haroldrandom) [inline src/virtual-wan/azext_vwan/tests/latest/ApplicationGatewayAuthCert.pem:20]: Is there any security concern to publish it ?

Comment (MyronFanQiu) [reply src/virtual-wan/azext_vwan/tests/latest/ApplicationGatewayAuthCert.pem:20]: I believe credscan 😄. It's also used in PowerShell.

File: src/virtual-wan/azext_vwan/_help.py (hunk 1 above)
Comment (MyronFanQiu) [reply src/virtual-wan/azext_vwan/_help.py:326]: add examples for create command

File: src/virtual-wan/azext_vwan/custom.py [hunk 6]
```diff
@@ -571,3 +577,239 @@ def update_vpn_site(instance, cmd, ip_address=None, virtual_wan=None, tags=None,
...
+                                    'P2SConnectionConfiguration',
+                                    'AddressSpace')
+    gateway = P2SVpnGateway(
+        location=location,
+        tags=tags,
+        virtual_hub=SubResource(id=virtual_hub) if virtual_hub else None,
+        vpn_gateway_scale_unit=scale_unit,
+        vpn_server_configuration=SubResource(id=vpn_server_config) if vpn_server_config else None,
+        p2_sconnection_configurations=[
+            P2SConnectionConfiguration(
```
Comment (MyronFanQiu) [inline src/virtual-wan/azext_vwan/custom.py:729]: here

**Pull request merged**